### PR TITLE
[NFC] Remove Intel Op codes that are being upstreamed

### PR DESCRIPTION
**Pull request by MrSidims**
Follows https://github.com/KhronosGroup/SPIRV-Headers/pull/298
Had to keep internal::DecorationRuntimeAlignedINTEL since the extension
implementation was done none conformant to the spec.

Signed-off-by: Sidorov, Dmitry <dmitry.sidorov@intel.com>

### CODE REVIEW

**Comment by MrSidims** [comment]
@asudarsa @vmaksimo please take a look

**Comment by MrSidims** [comment]
Found out, that runtime aligned feature is not conformant to the spec, had to keep the appropriate decoration in `internal `namespace

**Comment by MrSidims** [comment]
@asudarsa @vmaksimo friendly ping

File: lib/SPIRV/libSPIRV/SPIRVNameMapEnum.h [hunk 1]
```diff
@@ -182,10 +182,10 @@ template <> inline void SPIRVMap<Decoration, std::string>::init() {
...
   // From spirv_internal.hpp
   add(internal::DecorationCallableFunctionINTEL, "CallableFunctionINTEL");
-  add(internal::DecorationMathOpDSPModeINTEL, "MathOpDSPModeINTEL");
-  add(internal::DecorationInitiationIntervalINTEL, "InitiationIntervalINTEL");
-  add(internal::DecorationMaxConcurrencyINTEL, "MaxConcurrencyINTEL");
-  add(internal::DecorationPipelineEnableINTEL, "PipelineEnableINTEL");
+  add(DecorationMathOpDSPModeINTEL, "MathOpDSPModeINTEL");
+  add(DecorationInitiationIntervalINTEL, "InitiationIntervalINTEL");
+  add(DecorationMaxConcurrencyINTEL, "MaxConcurrencyINTEL");
+  add(DecorationPipelineEnableINTEL, "PipelineEnableINTEL");
```
Comment (vmaksimo) [inline lib/SPIRV/libSPIRV/SPIRVNameMapEnum.h:188]: It says "from spirv_internal.hpp", but as I understood the decorations are now in spirv.hpp header. Should we move this to other place?
(same for capabilities)

File: lib/SPIRV/libSPIRV/spirv_internal.hpp [hunk 2]
```diff
@@ -91,10 +84,7 @@ enum InternalExecutionMode {
   IExecModeStreamingInterfaceINTEL = 6154
 };
 
-enum InternalLoopControlMask {
-  ILoopControlLoopCountINTELMask = 0x1000000,
-  ILoopControlMaxReinvocationDelayINTELMask = 0x2000000
-};
+enum InternalLoopControlMask {};
```
Comment (vmaksimo) [inline lib/SPIRV/libSPIRV/spirv_internal.hpp:87]: Could you please clarify why we still need this trick?

Comment (MrSidims) [reply lib/SPIRV/libSPIRV/spirv_internal.hpp:87]: We don't need it, but we might add new Masks here in the future, so I don't want to remove enum

File: lib/SPIRV/libSPIRV/SPIRVNameMapEnum.h (hunk 1 above)
Comment (MrSidims) [reply lib/SPIRV/libSPIRV/SPIRVNameMapEnum.h:188]: Thanks! I'll move the code

File: lib/SPIRV/libSPIRV/spirv_internal.hpp (hunk 2 above)
Comment (vmaksimo) [reply lib/SPIRV/libSPIRV/spirv_internal.hpp:87]: I see your point, but it still looks strange to me. There is a chance that this line would not be found by a developer and a new enum will be created.

**Review by vmaksimo — COMMENTED** [review]
LGTM, but possibly need to resolve the discussion first.

File: lib/SPIRV/libSPIRV/spirv_internal.hpp (hunk 2 above)
Comment (vmaksimo) [reply lib/SPIRV/libSPIRV/spirv_internal.hpp:87]: I see your point, but it still looks strange to me. There is a chance that this line would not be found by a developer and a new enum will be created.

Comment (MrSidims) [reply lib/SPIRV/libSPIRV/spirv_internal.hpp]: Agree, removed the enum

**Review by vmaksimo — APPROVED** [review]

**Pull request merged**